libs/execution/src/lib/types/io-types/filesystem-inmemory.ts

```diff
@@ -4,94 +4,96 @@
 
 import { IOType } from '@jvalue/jayvee-language-server';
 
-import { BinaryFile } from './binary-file';
 import { FileSystem } from './filesystem';
+import { FileSystemDirectory } from './filesystem-node-directory';
+import { FileSystemFile } from './filesystem-node-file';
+import { BinaryFile } from './filesystem-node-file-binary';
 
 export class InMemoryFileSystem implements FileSystem {
   public readonly ioType = IOType.FILE_SYSTEM;
 
+  private rootDirectory: FileSystemDirectory = new FileSystemDirectory('root');
   private static PATH_SEPARATOR = '/';
   private static CURRENT_DIR = '.';
   private static PARENT_DIR = '..';
 
-  // Hierachical file system
-  private fileSystemIndex: Map<string, BinaryFile | FileSystem> = new Map();
+  getFile(path: string): FileSystemFile<unknown> | null {
```

Comment:
After refactoring the `findFile` method, it can be called here on `rootDirectory` to find the desired file.

Reply:
removed find file, so thats obsolet.